packages/ui/feature-dashboard/src/lib/pages/flows-table/flows-table-title/flows-table-title.component.ts

```diff
@@ -44,32 +62,97 @@ export class FlowsTableTitleComponent {
       FoldersSelectors.selectCurrentFolder
     );
   }
-  createFlow() {
+  createFlow(navigateAfterCreation: boolean, name?: string) {
     if (!this.createFlow$) {
+      if (navigateAfterCreation) {
+        this.builderService.showLoading();
+      }
       this.createFlow$ = this.currentFolder$.pipe(
         take(1),
         switchMap((res) => {
           return this.flowService
             .create({
               projectId: this.authenticationService.getProjectId(),
-              displayName: $localize`Untitled`,
+              displayName: name || $localize`Untitled`,
               folderId: res?.id,
             })
             .pipe(
               tap((flow) => {
-                localStorage.setItem(
-                  CURRENT_FLOW_IS_NEW_KEY_IN_LOCAL_STORAGE,
-                  'true'
-                );
-                this.router.navigate(['/flows/', flow.id]);
+                if (navigateAfterCreation) {
+                  this.builderService.hideLoading();
+                  this.router.navigate(['/flows/', flow.id]);
+                }
               })
             );
         })
       );
     }
+    return this.createFlow$;
+  }
+  openTemplatesDialog() {
+    const data: TemplateDialogData = {
+      insideBuilder: false,
+    };
+    this.openTemplatesDialog$ = this.matDialog
+      .open(TemplatesDialogComponent, { data })
+      .afterClosed()
+      .pipe(
+        switchMap((dialogResult?: TemplateDialogClosingResult) => {
+          if (dialogResult) {
+            this.builderService.showLoading();
+            return this.createFlow(false, dialogResult.template.name).pipe(
+              switchMap((flow) => {
+                return this.flowService.update(flow.id, {
+                  type: FlowOperationType.IMPORT_FLOW,
```

Comment:
Instead of creating then updating we need to have a single request that does both, will leave this like this for now, and add it to the next sprint.